### PR TITLE
windows compatibility changes

### PR DESCRIPTION
**Pull request by hansieodendaal**
## Description
Windows build and test compatibility

## Motivation and Context
Test files cleanup
- In Windows, files cannot be deleted if the filehandles are not closed beforehand. Added scope brackets to tests where required.

Line endings
- Git manages line endings on Windows vs. Linix, so `cargo-fmt --all` needs to use the `newline_style = "Native"` setting to be compatible between the two systems.
- Needed to pass platform-agnostic line ending pattern to the `std::string::String split()` method to read test data

## How Has This Been Tested?
- Tests for all changed components passed on Windows 10. Issues with `tari_wallet` still need to be sorted out. probably due to C dependencies.
- Tests passed in Linux Ubuntu.

## Types of changes

* [X] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [X] I'm merging against the `development` branch
* [X] I ran `cargo-fmt --all` before pushing
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

File: rustfmt.toml [hunk 1]
```diff
@@ -7,7 +7,7 @@ imports_layout = "HorizontalVertical"
 merge_imports = true
 match_block_trailing_comma = true
 max_width = 120
-newline_style = "Unix"
+newline_style = "Native"
```
Comment (sdbondi) [inline rustfmt.toml]: Was `Unix` newlines causing an issue for you? I'd think it's better to have consistent newlines (i.e always `\n`) for all platforms. I think git will correct for this in either case.

File: infrastructure/storage/tests/lmdb.rs [hunk 2]
```diff
@@ -94,11 +94,16 @@ fn clean_up(name: &str) {
     std::fs::remove_dir_all(get_path(name)).unwrap();
 }
 
+#[cfg(windows)]
+const LINE_ENDING: &'static str = "\r\n";
+#[cfg(not(windows))]
+const LINE_ENDING: &'static str = "\n";
```
Comment (sdbondi) [inline infrastructure/storage/tests/lmdb.rs]:
```suggestion
#[cfg(windows)]
const LINE_ENDING: &str = "\r\n";
#[cfg(not(windows))]
const LINE_ENDING: &str = "\n";
```

File: rustfmt.toml (hunk 1 above)
Comment (hansieodendaal) [reply rustfmt.toml]: This is a problem for `cargo-fmt --all` as git automatically convert line endings to Windows-style when checking it out from GitHub. So running `cargo-fmt --all` changes all line endings in all files.

Comment (sdbondi) [reply rustfmt.toml]: Ok makes sense

File: infrastructure/storage/tests/lmdb.rs [hunk 3]
```diff
@@ -117,136 +122,150 @@ fn insert_all_users(name: &str) -> (Vec<User>, LMDBDatabase) {
...
+        // Test the `exists` and value retrieval functions
+        let res = db.with_read_transaction::<_, User>(|txn| {
+            for user in users.iter() {
+                assert!(txn.exists(&user.id).unwrap());
+                let check: User = txn.get(&user.id).unwrap().unwrap();
+                assert_eq!(check, *user);
+            }
+            Ok(None)
+        });
+        println!("{:?}", res);
```
Comment (CjS77) [inline infrastructure/storage/tests/lmdb.rs]: You can remove this

**Pull request merged**